src/harness/util.py

```diff
@@ -45,76 +52,67 @@ def _getRandomLatLongInPolygon(polygon):
     else:
-      return _getRandomLatLongInPolygon(polygon)
+      return getRandomLatLongInPolygon(ppa)
   except NotImplementedError:
-    # If there's no intercept again call the function
-    return _getRandomLatLongInPolygon(polygon)
-
-
-def _changeDateInPpaAndPal(record, prev_date, next_date):
-  for key, value in record.iteritems():
-    if type(record[key]) is dict:
-      _changeDateInPpaAndPal(record[key], prev_date, next_date)
-    record[key] = (next_date if 'Expiration' in key else
-                  prev_date if 'Date' in key else record[key])
-  return record
+    # Cannot get the intercept call it again
+    return getRandomLatLongInPolygon(ppa)
 
 
-def makePpaAndPalRecordsConsistent(ppa_record, pal_records,
-                                   ppa_fips_code, ppa_census_year,
-                                   low_frequency, high_frequency, device):
+def makePpaAndPalRecordsConsistent(ppa_record, pal_records, low_frequency,
+                                   high_frequency, user_id):
   """Make PPA, PAL and Device object consistent with the inputs and position 
   the device in the PPA Polygon at some random location
 
     Args:
-      ppa_record: (dictionary) An object containing PPA Record.
-      pal_records: (list) A list of PAL Records which has to be
-        associated with the PPA.
-      ppa_fips_code: (number) The FIPS Code of the Census Tract where 
-      the PPA belongs.
-      ppa_census_year: (number) The census year in which the census 
-      tract was defined.
-      low_frequency: (number) The Primary Low Frequency for PAL
-      high_frequency: (number) The Primary High Frequency for PAL
-      device: (dictionary) An object containing device which has to be
-      moved into the PPA Polygon.
+      ppa_record: (dictionary) A dictionary containing PPA Record.
+      pal_records: (list) A list of PAL Records in the form of dictionary
+      which has to be associated with the PPA.
+      low_frequency: (number) The Primary Low Frequency for PAL.
+      high_frequency: (number) The Primary High Frequency for PAL.
+      user_id: (string) The userId from the CBSD.
+
     Returns:
-      A tuple consisting of PPA Record, PAL Record List and Device with 
-      modified Latitude and Longitude
+      A tuple containing pal records list which contains individual pal records 
+      in the form of dictionary and a ppa record which itself is a dictionary.
+    Note: The PPA Dictionary must contain censusYear (number) and fipsCode(number)
   """
 
   previous_year_date = datetime.now().replace(year=datetime.now().year - 1)
   next_year_date = datetime.now().replace(year=datetime.now().year + 1)
+  fcc_channel_id = '1'
+  ppa_fips_code = ppa_record['fipsCode']
+  ppa_census_year = ppa_record['censusYear']
+  del ppa_record['censusYear'], ppa_record['fipsCode']
 
-  for pal_record in pal_records:
+  for index, pal_rec in enumerate(pal_records):
+    pal_rec = defaultdict(lambda: defaultdict(dict), pal_rec)
```

Comment:
This helps to add a key into the dictionary without raising the KeyError of some default type. For example in case of **pal_rec['channelAssignment']['primaryAssignment']['lowFrequency']** it will raise a KeyError as channelAssignment does not exist in the dictionary. To get rid of a lot of ifs to check if Key is there or not we can use defaultdict which will create a key in the dictionary with some default type if it's not there.